bundles/org.openhab.binding.sensibo/src/main/java/org/openhab/binding/sensibo/internal/handler/SensiboSkyHandler.java

```diff
@@ -0,0 +1,445 @@
+                        addDynamicChannelsAndProperties(sensiboSky);
+                    }
+                } else if (CHANNEL_SWING_MODE.equals(channelUID.getId())) {
+                    if (command instanceof RefreshType && sensiboSky.getAcState().isPresent()) {
+                        if (sensiboSky.getAcState().isPresent() && sensiboSky.getAcState().get().getSwing() != null) {
+                            updateState(channelUID, new StringType(sensiboSky.getAcState().get().getSwing()));
+                        } else {
+                            updateState(channelUID, UnDefType.UNDEF);
+                        }
+                    } else if (command instanceof StringType) {
+                        final StringType newValue = (StringType) command;
+                        updateAcState(sensiboSky, SWING_PROPERTY, newValue.toString());
+                    }
+                } else if (CHANNEL_FAN_LEVEL.equals(channelUID.getId())) {
+                    if (command instanceof RefreshType) {
+                        if (sensiboSky.getAcState().isPresent()
+                                && sensiboSky.getAcState().get().getFanLevel() != null) {
+                            updateState(channelUID, new StringType(sensiboSky.getAcState().get().getFanLevel()));
+                        } else {
+                            updateState(channelUID, UnDefType.UNDEF);
+                        }
+                    } else if (command instanceof StringType) {
+                        final StringType newValue = (StringType) command;
+                        updateAcState(sensiboSky, FAN_LEVEL_PROPERTY, newValue.toString());
+                    }
+                } else if (CHANNEL_TIMER.equals(channelUID.getId())) {
+                    if (command instanceof RefreshType) {
+                        if (sensiboSky.getTimer().isPresent() && sensiboSky.getTimer().get().secondsRemaining > 0) {
+                            updateState(channelUID, new DecimalType(sensiboSky.getTimer().get().secondsRemaining));
+                        } else {
+                            updateState(channelUID, UnDefType.UNDEF);
+                        }
+                    } else if (command instanceof DecimalType) {
+                        final DecimalType newValue = (DecimalType) command;
+                        updateTimer(sensiboSky, newValue.intValue());
+                    } else {
+                        updateTimer(sensiboSky, null);
+                    }
+                }
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "Unreachable by Sensibo servers");
+            }
+        });
+    }
+
+    @Override
+    public Collection<Class<? extends ThingHandlerService>> getServices() {
+        return Collections.unmodifiableList(Stream.of(CallbackChannelsTypeProvider.class).collect(Collectors.toList()));
```

Comment:
Are you expecting more services to be added later? Why not just replace this with `return Collections.singleton(CallbackChannelsTypeProvider.class);`?

Reply:
No not really. Fixed.